server/server.go

```diff
@@ -43,5 +39,32 @@ func Serve(port int, allowIP string) error {
 		connCh <- conn
 		rpc.ServeConn(conn)
 	}
-	return nil
+}
+
+// ServeLocal is to fall back when lemonade client can't connect server.
+// returns port number, error
+func ServeLocal() (int, error) {
+	l, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		return 0, err
+	}
+	go func() {
+		for {
+			conn, err := l.Accept()
+			if err != nil {
+				log.Panicln(err)
+				continue
+			}
+			connCh <- conn
+			rpc.ServeConn(conn)
+		}
+	}()
+	return l.Addr().(*net.TCPAddr).Port, nil
+}
+
+func init() {
+	uri := &URI{}
+	rpc.Register(uri)
```

Comment:
__[gometalinter]__
* [errcheck] error return value not checked (rpc.Register(uri))